modules/axtask/src/wait_queue.rs

```diff
@@ -117,12 +116,14 @@ impl WaitQueue {
         );
         crate::timers::set_alarm_wakeup(deadline, curr.clone());
 
-        rq.blocked_resched(self.queue.lock());
+        rq.blocked_resched(self.queue.lock(), waiter.clone());
 
-        let timeout = curr.in_wait_queue(); // still in the wait queue, must have timed out
+        let timeout = axhal::time::wall_time() >= deadline;
 
+        // Always try to remove the task from wait list.
+        self.queue.lock().remove(&waiter);
         // Always try to remove the task from the timer list.
-        self.cancel_events(curr, true);
+        self.cancel_timer(curr);
```

Comment:
can we keep the `cancel_events` and move `self.queue.lock().remove(&waiter);` into it

Reply:
good idea,  to previous version.